lib/src/script.dart

```diff
@@ -479,8 +543,13 @@ class Script {
   ///
   /// If [silenceStartMessage] is `false` (the default), this prints a message
   /// in debug mode indicating that the script has started running.
-  Script._(this.name, StreamConsumer<List<int>> stdin, Stream<List<int>> stdout,
-      Stream<List<int>> stderr, Future<int> exitCode,
+  Script._(
+      this.name,
+      StreamConsumer<List<int>> stdin,
+      Stream<List<int>> stdout,
+      Stream<List<int>> stderr,
+      Future<bool Function(ProcessSignal)> signalHandler,
```

Comment:
I think passing this in as a future is wrong, for a few reasons:

1. This being a future means that there's a gap between creating a `Script` and it fully initializing when `Script.kill()` won't do anything.

2. It makes sense to be able to kill a `Script` while it's initializing. For example, in `Script()`, there's an async gap when evaluating the script's arguments—if the script is killed then, it should avoid spawning the subprocess entirely.

3. Making this a non-`Future` avoids a lot of complexity around around completers and a lot of cases where you've otherwise needed to change `Script._()` calls to `Script.fromCompleter()`.